### PR TITLE
Add VNET rules to PostgreSQL

### LINKED ISSUE #667: Add vnet rules to postgresql module
It would be very useful to be able to create vnet rules for the provisioned db just as it can be done using the portal and cli.

### PR DESCRIPTION
**Pull request by timja**
Closes #667

Manually tested:

* azure-postgresql-10-dbms
* azure-postgresql-10

Added life cycle test coverage as well

### CODE REVIEW

**Comment by timja** [comment]
Is it expected that lint should take greater than 10 minutes locally?

```
$ docker ps
CONTAINER ID        IMAGE                                       COMMAND                  CREATED             STATUS              PORTS                    NAMES
18c8f9e10e0e        quay.io/deis/lightweight-docker-go:v0.2.0   "gometalinter ./... …"   11 minutes ago      Up 11 minutes                                sad_almeida
```

Tests etc executed at a reasonable speed for me, but lint was ridiculous, I thought it wasn't working so kept cancelling it

**Comment by zhongyi-zhang** [comment]
Not expected to be so long. I kicked off the CI. The lint test result looks good. Again thanks for the contribution! I'll look into the code change later.

File: pkg/services/postgresql/plan_schemas.go [hunk 1]
```diff
@@ -128,6 +128,28 @@ func generateUpdatingParamsSchema(
 					},
 				},
 			},
+			"virtualNetworkRules": &service.ArrayPropertySchema{
+				Title:       "Virtual network rules",
+				Description: "Virtual network rules to apply to instance. ",
+				ItemsSchema: &service.ObjectPropertySchema{
+					Title:       "Virtual network rule",
+					Description: "Individual virtual network rules",
```
Comment (zhongyi-zhang) [inline pkg/services/postgresql/plan_schemas.go]: `rules` -> `rule`.

File: pkg/services/postgresql/dbms_arm_template.go [hunk 2]
```diff
@@ -57,7 +58,22 @@ var dbmsARMTemplateBytes = []byte(`
...
+					"type": "virtualNetworkRules",
+					"apiVersion": "[variables('DBforPostgreSQLapiVersion')]",
+					"dependsOn": [
+						"Microsoft.DBforPostgreSQL/servers/{{ $.serverName }}"
+					],
+					"location": "{{$root.location}}",
+					"name": "{{$rule.name}}",
+					"properties": {
+						"ignoreMissingVnetServiceEndpoint": true,
+        				"virtualNetworkSubnetId": "{{$rule.subnetId}}"
```
Comment (zhongyi-zhang) [inline pkg/services/postgresql/dbms_arm_template.go]: indent

Comment (timja) [reply pkg/services/postgresql/dbms_arm_template.go]: something weird going on with the indentation in VSCode,
After fixing it in `vim` it looks like this in vscode =/

image

fixed anyway

**Pull request merged**